api/src/main/java/org/openmrs/api/ConceptService.java

```diff
@@ -1922,5 +1922,57 @@ public Drug getDrugByMapping(String code, ConceptSource conceptSource,
 	 */
 	@Authorized(PrivilegeConstants.GET_CONCEPTS)
 	boolean hasAnyConceptAttribute(ConceptAttributeType conceptAttributeType);
-	
+	/**
+	 * Get Concept by id or name convenience method
+	 * 
+	 * @param idOrMap
+	 * @return the found Concept
+	 * @throws APIException
+	 */
+	@Authorized(PrivilegeConstants.GET_CONCEPTS)
+	public static Concept getConceptInfo(String idOrMap) throws APIException {
+
+		Concept cpt = null;
+
+		if (idOrMap != null) {
+
+			idOrMap = idOrMap.trim();
+
+			//if input contains at least one period handle recursively as a code constant
+			if (idOrMap.contains(".")) {
+				return getConcept(evaluateStaticConstant(idOrMap));
+			}
+
+			// handle  mapping id: xyz:ht
+			int index = id.indexOf(":");
```

Comment:
not sure why you are treating the input as a Map. According to your method description it is either **an ID or a Name**. Isn't it?

Reply:
How best can i include a map as an input?